### PR TITLE
Add test for spending more gas on L2 than allotted on L1

### PR DESCRIPTION
**Pull request by mslipper**
Fixes ENG-2384

### CODE REVIEW

File: contracts/Failure.sol [hunk 1]
```diff
@@ -5,4 +5,12 @@ contract Failure {
     function fail() public payable {
         revert("big fail!");
     }
+
+    function failWithBurn(uint256 _amount) public payable {
```
Comment (protolambda) [inline contracts/Failure.sol]: If the amount is smaller than the provided gas it doesn't fail. Maybe just rename it to `burnGas` and add a docstring it burns at least that much gas? (but not exactly)

File: simulators/optimism/l1ops/deposit_tests.go [hunk 2]
```diff
@@ -299,6 +299,12 @@ func failingDepositWithMintTest(t *hivesim.T, env *optimism.TestEnv) {
 	revertTx, err := failureContract.Fail(l2Opts)
 	require.NoError(t, err)
 
+	// Create transaction that burns more gas on L2 than allotted
+	// in the deposit transaction.
+	l2Opts.GasLimit = 3_000_000
+	burnTx, err := failureContract.FailWithBurn(l2Opts, big.NewInt(2_000_000))
```
Comment (protolambda) [inline simulators/optimism/l1ops/deposit_tests.go]: At 2M burned, and 3M provided, isn't it supposed to pass without exceeding the tx gas limit?

Comment (mslipper) [reply simulators/optimism/l1ops/deposit_tests.go]: Doesn't the gas limit get pulled from the `DepositTransaction` call below?

Comment (mslipper) [reply simulators/optimism/l1ops/deposit_tests.go]:
```
tx, err := depositContract.DepositTransaction(
			opts,
			depositor,
			mintAmount,
			1_000_000,
			false,
			data,
		)
```

Comment (protolambda) [reply simulators/optimism/l1ops/deposit_tests.go]: Oh I see, you're preparing the tx here, but then copying the data and repeating it with a lower limit, nice.

File: contracts/Failure.sol (hunk 1 above)
Comment (mslipper) [reply contracts/Failure.sol]: Renamed to `burn`.

File: simulators/optimism/p2p/main.go [hunk 3]
```diff
@@ -3,12 +3,13 @@ package main
...
 	"fmt"
-	"github.com/ethereum/go-ethereum/core/types"
-	"github.com/ethereum/go-ethereum/params"
 	"math/big"
 	"sync"
 	"time"
 
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/params"
+
```
Comment (trianglesphere) [inline simulators/optimism/p2p/main.go]: extra blank space here.

File: simulators/optimism/l1ops/main.go [hunk 4]
```diff
@@ -10,10 +10,10 @@ import (
 )
 
 var tests = []*optimism.TestSpec{
-	{Name: "deposit simple tx through the portal", Run: simplePortalDepositTest},
-	{Name: "deposit contract creation through the portal", Run: contractPortalDepositTest},
-	{Name: "erc20 roundtrip through the bridge", Run: erc20RoundtripTest},
-	{Name: "simple withdrawal", Run: simpleWithdrawalTest},
+	//{Name: "deposit simple tx through the portal", Run: simplePortalDepositTest},
```
Comment (trianglesphere) [inline simulators/optimism/l1ops/main.go]: intentional to skip these?

**Review by trianglesphere — COMMENTED** [review]
test logic + new wait receipt looks good.

**Review by protolambda — APPROVED** [review]
See comment from Josh, need to re-enable other tests. LGTM otherwise.

**Comment by mslipper** [comment]
Derp, that was sloppy. Fixing.

File: simulators/optimism/l1ops/main.go (hunk 4 above)
Comment (mslipper) [reply simulators/optimism/l1ops/main.go]: Nope, nice catch.

**Review by trianglesphere — APPROVED** [review]
Lgtm

**Pull request merged**